x-pack/platform/plugins/shared/ml/public/application/explorer/explorer_utils.ts

```diff
@@ -700,3 +705,28 @@ export async function getDataViewsAndIndicesWithGeoFields(
   }
   return { sourceIndicesWithGeoFieldsMap, dataViews: [...dataViewsMap.values()] };
 }
+
+// Creates index pattern in the format expected by the kuery bar/kuery autocomplete provider
+// Field objects required fields: name, type, aggregatable, searchable
+export function getIndexPattern(influencers: ExplorerJob[]) {
```

Comment:
Might also be worth adding some simple jest unit tests for `getIndexPattern` and `getMergedGroupsAndJobsIds` while you're at it 😅 .